torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1591,14 +1591,6 @@ void ProcessGroupNCCL::watchdogHandler() {
           }
         }
 
-        if (SHOULD_CLEAN_UP(asyncErrorHandling_)) {
-          // Abort work and corresponding communicators
-          work.abort();
```

Comment:
after looking at the code for `work.abort()`

```
  // Abort all communicators of this work
  ncclComm_->ncclCommAbort();

  ncclCommDevIdxMapMutex.lock();
  ncclCommDevIdxMap.erase(ncclComm_);
  ncclCommDevIdxMapMutex.unlock();
```

And the code for processgroup::abort()

```
  ncclCommDevIdxMapMutex.lock();
  for (auto& it : devNCCLCommMap_) {
    auto& ncclComm = it.second;
    ncclCommDevIdxMap.erase(ncclComm);
  }
  ncclCommDevIdxMapMutex.unlock();

  std::lock_guard<std::mutex> lock(mutex_);
  abortCommsFromMap(devNCCLCommMap_, abortReason);
  abortCommsFromMap(inInitializationCommMap_, abortReason);
```

It looks like indeed the work.abort is a subset of the processgroup abort, as long as 2 things are true
 1. the work's communicator is in either `devNCCLCommMap_` or `inInitializationCommMap_`
 2. the ordering of the aborts doesn't matter.  (its not important to abort the communicator for the bad work obj before aborting other communicators

@eqy @minsii @kwen2501 any input on that?

Reply:
1. First one is always true, otherwise that's a bug
2. We only have one device per PG now, so there is really no ordering problem within a single PG

Reply:
for (2) i think we can still have multiple communicators active on one device/process. The ordering of aborting these communicators is what I was referring to.  (I have no idea if there is a case where this matters, just pointing it out).

Reply:
My understanding is that, for one process, if you have multiple PGs, you can have multiple comms. But  devNCCLCommMap_ is defined within a PG, after Ke's refactoring, there should be only one device per PG, and hence one comm per PG. @kwen2501, Is my understanding correct?

Reply:
> for (2) i think we can still have multiple communicators active on one device/process

This is true. 
A "legacy case" relates to how we create P2P comms for a rank. Instead of using a global comm this rank belong too, ProcessGroupNCCL would create a two-rank comm with the peer this rank would talk to. So that creates a possibility that a rank can have multiple comms, even if it owns one device only now.